### PR TITLE
Component/4221 icon

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

#4221 

Smaller and default
image

Larger and colored
image

Storybook options
image

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Screenshots/Recordings**

_If applicable, add screenshots or recordings to visualize the changes_

**Issue**

Progresses #???

### CODE REVIEW

File: app/component-library/components/Icon/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+### `size`
+
+Enum to select between size variants.
+
+| <span style="color:gray;font-size:14px">TYPE</span>    | <span style="color:gray;font-size:14px">REQUIRED</span> |
+| :----------------------------------------------------- | :------------------------------------------------------ |
+| [BaseAvatarSize](../BaseAvatar/BaseAvatar.types.ts#L6) | Yes                                                     |
+
+### `iconName`
```
Comment (Cal-L) [inline app/component-library/components/Icon/README.md:17]: Fix this to be called "name"

File: app/component-library/hooks/useStyles.ts [hunk 2]
```diff
@@ -13,11 +13,11 @@ import { Theme } from '../../util/theme/models';
...
   vars: V,
-): R => {
+): { styles: R; theme: Theme } => {
   const theme = useAppThemeFromContext();
   const styles = useMemo(
     () => styleSheet({ theme, vars }),
     [styleSheet, theme, vars],
   );
-  return styles;
+  return { styles, theme };
```
Comment (Cal-L) [inline app/component-library/hooks/useStyles.ts:22]: Better to split out this into its own PR to remove noise from this one

File: metro.config.js [hunk 3]
```diff
@@ -6,22 +6,26 @@
...
 
-module.exports = {
-  transformer: {
-    getTransformOptions: async () => ({
-      transform: {
-        experimentalImportSupport: true,
-        inlineRequires: true,
-      },
-    }),
-    assetPlugins: ['react-native-svg-asset-plugin'],
```
Comment (Cal-L) [inline metro.config.js:17]: May have to add this back in until token detection changes are in.

**Review by jpcloureiro — APPROVED** [review]
LGTM! 🎉
